### PR TITLE
Closes #23

### LINKED ISSUE #23: `DataMixture` clean up
To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] for classes:
    - [x] test default constructor
    - [x] for reference classes: test methods
  - [x] for functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] keep externally used names so that we don't break downstream code
  - [x] consider renaming internal variables etc. if needed
  - [x] if a function is too long, break up into pieces
  - [x] for classes: move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
  - [x] in `setClass` function, just use the `slots` instead of outdated `representation`.
  - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #23

### CODE REVIEW

File: R/Data-class.R [hunk 1]
```diff
@@ -178,151 +179,126 @@ DataDual <- function(w = numeric(),
...
+#'   for the share patients.
+#' @param ... parameters passed to [Data()].
+#'
+#' @export
+#' @example examples/Data-class-DataMixture.R
+DataMixture <- function(xshare = numeric(),
+                        yshare = integer(),
+                        ...) {
+  assert_numeric(xshare)
+  assert_integer(yshare, lower = 0, upper = 1)
```
Comment (danielinteractive) [inline R/Data-class.R:292]:
```suggestion
```

File: tests/testthat/helper-data.R [hunk 2]
```diff
@@ -25,3 +25,33 @@ h_get_data_dual <- function() {
...
+    doseGrid = d@doseGrid,
+    placebo = d@placebo,
+    ID = d@ID,
+    cohort = d@cohort
+  )
+}
+
+h_get_data_mixture <- function() {
+  d <- h_get_data()
+  DataMixture(
```
Comment (danielinteractive) [inline tests/testthat/helper-data.R:46]: can use `.DataMixture(d, ...)`

File: R/Data-validity.R [hunk 3]
```diff
@@ -113,3 +113,52 @@ validate_data_dual <- function(object) {
...
+
+#' @describeIn validate_data_objects validates that the [`DataMixture`] object
+#' contains valid elements with respect to their types, dependency and length.
+validate_data_mixture <- function(object) {
+  o <- Validate()
+  # This below check is a bit obsolete as below invocations to
+  # test_* throw an error when len parameter (here initiated to object@nObsshare)
+  # is not a scalar.
+  o$check(
+    test_int(object@nObsshare),
```
Comment (danielinteractive) [inline R/Data-validity.R:148]: let's use if else and return early if not ok

File: R/Data-validity.R [hunk 4]
```diff
@@ -16,10 +16,10 @@ NULL
 #'   `ID` and `cohort` vectors which are of the same length `nObs`.
 validate_subjects <- function(object) {
   o <- Validate()
-  o$check(
-    test_int(object@nObs),
-    "nObs must be of type integer of length 1"
-  )
+  if (!test_int(object@nObs)) { # in if clause so that below test_* won't fail
```
Comment (danielinteractive) [inline R/Data-validity.R:19]: longer comments better in separate line above, and please use sentence style (so start with upper case letter, end with full stop). thx

File: R/Data-validity.R [hunk 5]
```diff
@@ -16,10 +16,10 @@ NULL
...
+    o$check(FALSE, "nObs must be of type integer of length 1")
+    return(o$result())
```
Comment (danielinteractive) [inline R/Data-validity.R:21]: here I would directly put the simpler
```suggestion
    return("nObs must be of type integer of length 1")
```

File: R/Data-validity.R [hunk 6]
```diff
@@ -136,3 +136,27 @@ validate_data_parts <- function(object) {
...
+validate_data_mixture <- function(object) {
+  o <- Validate()
+
+  o$check(
+    test_subset(object@xshare, object@doseGrid),
+    "Dose values in xshare must be from doseGrid"
+  )
+  if (!test_int(object@nObsshare)) { # in if clause so that below test_* won't fail
+    o$check(FALSE, "nObsshare must be of type integer of length 1")
+    return(o$result())
```
Comment (danielinteractive) [inline R/Data-validity.R:151]: same comment here

File: tests/testthat/test-Data-validity.R [hunk 7]
```diff
@@ -157,3 +157,62 @@ test_that("validate_data_parts returns error for part1Ladder with wrong values",
...
+
+  # We assign nObsshare of length different than 1,
+  object@nObsshare <- 1:5
+
+  # validate_data_mixture will throw the error here, which will be originated by
+  # test_* functions (used in validate_data_mixture) when the len parameter
+  # (initiated to object@nObsshare) is not a scalar.
+  expect_error(
+    validate_data_mixture(object),
+    regexp = "^Argument '[[:alnum:]]+' must have length 1$"
```
Comment (danielinteractive) [inline tests/testthat/test-Data-validity.R:216]: this should no longer be the case now?

**Review by danielinteractive — COMMENTED** [review]
thx - please see inside some simplification/formatting and question on test

File: R/Data-validity.R (hunk 5 above)
Comment (wwojciech) [reply R/Data-validity.R:21]: Sure, but that would assume that there will be no tests defined before this check.....

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**